File: renu_customization/public/js/hide_fields_on_picklist.js
frappe.ui.form.on('Pick List', {
    onload: function(frm) {
        // Hide fields
        frm.set_df_property('sales_order', 'hidden', 1);
        frm.set_df_property('invoice_type', 'hidden', 1);
    }
});

File: renu_customization/patches/v_0/hide_fields_on_picklist.py
import frappe

def execute():
    # Fields to hide in Pick List
    fields_to_hide = ["sales_order", "invoice_type"]

    for fieldname in fields_to_hide:
        custom_field = frappe.db.exists("Custom Field", {"dt": "Pick List", "fieldname": fieldname})
        if custom_field:
            frappe.db.set_value("Custom Field", custom_field, "hidden", 1)

    # Clear cache for Pick List
    frappe.clear_cache(doctype="Pick List")
